packages/frontend/src/features/catalog/components/CatalogPanel.tsx

```diff
@@ -109,43 +124,68 @@ export const CatalogPanel: FC<React.PropsWithChildren<Props>> = ({
                 'ArrowDown',
                 () => {
                     //FIXME fix bug when multiple "fields" have the same name in search and you can't move
-                    if (selectedTable) {
+                    if (selection) {
                         //TODO move around grouped items
-                        const selectedIndex = ungroupedCatalogItems.findIndex(
-                            (item) => item.name === selectedTable,
+                        //TODO also, this sort could go somewhere else
+                        const sortedGroupTables = Object.keys(
+                            catalogTree[selection.group].tables,
+                        ).sort(([a], [b]) => a.localeCompare(b));
+
+                        const selectedIndex = sortedGroupTables.findIndex(
+                            (name) => name === selection.table,
                         );
                         if (
                             selectedIndex !== undefined &&
-                            selectedIndex < ungroupedCatalogItems.length
+                            selectedIndex < sortedGroupTables.length
                         ) {
                             selectAndGetMetadata(
-                                ungroupedCatalogItems[selectedIndex + 1].name,
+                                sortedGroupTables[selectedIndex + 1],
+                                selection.group,
                             );
                         }
-                    } else selectAndGetMetadata(ungroupedCatalogItems[0]?.name);
+                    } else
+                        selectAndGetMetadata(
+                            catalogTree[Object.keys(catalogTree)[0]].tables[0]
+                                .name,
+                            Object.keys(catalogTree)[0],
+                        );
                 },
             ],
             [
                 'ArrowUp',
                 () => {
-                    if (selectedTable) {
+                    if (selection) {
                         //TODO move around grouped items
-                        const selectedIndex = ungroupedCatalogItems.findIndex(
-                            (item) => item.name === selectedTable,
+                        const sortedGroupTables = Object.keys(
```

Comment:
we could extract some of this logic outside the `arrow` methods and reuse

Reply:
Yea, probably a good call. Im actually thinking maybe the sort should happen on tree build...

Reply:
I made it so the sort happens on the tree build. It adds more computation to building the tree, but we were doing that here and in render anyway.